File: superstitioApi/java/superstitioapi/hangUpCard/CardOrb_WaitCardTrigger.java
package superstitioapi.hangUpCard;

import com.badlogic.gdx.graphics.Color;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.CardGroup;
import com.megacrit.cardcrawl.orbs.AbstractOrb;
import org.apache.logging.log4j.util.BiConsumer;
import superstitioapi.utils.ActionUtility;

public class CardOrb_WaitCardTrigger extends CardOrb_CardTrigger implements ICardOrb_WaitTime {

    protected static final Color ReduceWaitTime = Color.GOLDENROD.cpy();

    public CardOrb_WaitCardTrigger(AbstractCard card, CardGroup cardGroupReturnAfterEvoke, int waitTime, BiConsumer<CardOrb_CardTrigger,
            AbstractCard> action_thisOrb_triggerCard) {
        super(card, cardGroupReturnAfterEvoke, waitTime, action_thisOrb_triggerCard);
    }

    @Override
    public AbstractOrb makeCopy() {
        return new CardOrb_WaitCardTrigger(getOriginCard(), cardGroupReturnAfterEvoke, OrbCounter, action);
    }

    @Override
    protected ActionUtility.VoidSupplier checkAndSetTheHoverType() {
        if (this.OrbCounter > 1)
            return this::State_WhenHoverCard_JustGlow;
        return super.checkAndSetTheHoverType();
    }

    private void State_WhenHoverCard_JustGlow() {
        this.card.targetDrawScale = DRAW_SCALE_SMALL_BIGGER;
        this.card.glowColor = ReduceWaitTime;
    }

    @Override
    protected boolean onProperCardUsed_checkIfShouldApplyAction(AbstractCard card) {
        return OrbCounter <= 0;
    }
}

File: superstitioApi/java/superstitioapi/hangUpCard/CardOrb_WaitCardTrigger.kt
package superstitioapi.hangUpCard

import com.badlogic.gdx.graphics.Color
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.CardGroup
import com.megacrit.cardcrawl.orbs.AbstractOrb
import superstitioapi.utils.ActionUtility.VoidSupplier
import superstitioapi.utils.CostSmart

class CardOrb_WaitCardTrigger(
    card: AbstractCard,
    cardGroupReturnAfterEvoke: CardGroup?,
    waitTime: CostSmart,
    action_thisOrb_triggerCard: (CardOrb_CardTrigger, AbstractCard) -> Unit
) : CardOrb_CardTrigger(card, cardGroupReturnAfterEvoke, waitTime, action_thisOrb_triggerCard), ICardOrb_WaitTime
{
    private fun State_WhenHoverCard_JustGlow()
    {
        fakeCard.targetDrawScale = DRAW_SCALE_SMALL_BIGGER
        fakeCard.glowColor = ReduceWaitTime
    }

    override fun makeCopy(): AbstractOrb
    {
        return CardOrb_WaitCardTrigger(originCard, cardGroupReturnAfterEvoke, orbCounter, action)
    }

    override fun checkAndSetTheHoverType(): VoidSupplier
    {
        if (!this.orbCounter.isZero())
            return VoidSupplier(this::State_WhenHoverCard_JustGlow)
        return super.checkAndSetTheHoverType()
    }

    override fun onProperCardUsed_checkIfShouldApplyAction(card: AbstractCard?): Boolean
    {
        return orbCounter.isZero()
    }

    companion object
    {
        protected val ReduceWaitTime: Color = Color.GOLDENROD.cpy()
    }
}
